### DISCUSSION TITLE
Can't run snap7 without an environment

### DISCUSSION THREAD

**Discussion by FernandoBarajasValdivia** [gijzelaerr/python-snap7#440] [Q&A] [answered]
I installed snap7 in a jetson nano,  using an environmet and it is all okey.

The problem is when i install snap7 without an environment I get next error:

snap7.exceptions.snap7Exceptions: can't find snap7 library. If installed, try running Idconfig

(The install was successful but I can't use it for the error)

**Comment by nikteliy** [comment]
Hi @FernandoBarajasValdivia 
Can you give more information about your setup?
1. What version of Python are you using? (command: `python3 -V`)
2. What version of the package is installed? (command: `python3 -m pip show python-snap7`)
3. What package manager are you using? (pip, poetry, anaconda)

**Reply by FernandoBarajasValdivia** [reply]
1.- python version= 3.6.9
2.- version of the package:
![image](https://user-images.githubusercontent.com/99165285/229991232-57620a8d-20c6-4a27-bb6a-6d92bafe23a7.png)
3.- package manager: pip3

thanks for response!

**Reply by nikteliy** [reply]
I've checked a package build for v1.2, looks fine to me.
1. Show your OS version with `uname -a` command
2. Check if the shared library exists using the `ls /home/jetsonoc/.local/lib/python3.6/site-packages/snap7/lib/libsnap7.so` command

**Comment by gijzelaerr** [comment]
are you sure you are running on 64 bit arm? can you try to manually install the binary arm64 wheel from pypi

https://files.pythonhosted.org/packages/ea/14/6dda70653bcfd91010cccf5488f80c0fe9551baf8c549f5aab1eef1eb945/python_snap7-1.3-py3-none-manylinux2014_aarch64.whl

**Reply by FernandoBarajasValdivia** [reply]
when I try to install python_snap7-1.3-py3-none-manylinux2014_aarch64.whl I receive an error like this:
"python_snap7-1.3-py3-none-manylinux2014_aarch64.whl is not a supported wheel on this plataform" :(

**Reply by gijzelaerr** [reply]
is your platform 32 or 64 bit

**Comment by Rahul14189** [comment]
I have got the similar type issue, while installing it and running through docker, i have installed it using "pip install python-snap7" in docker file, this is my error

File "/py/lib/python3.9/site-packages/snap7/common.py", line 63, in __init__
logic-load               |     raise RuntimeError("can't find snap7 library. If installed, try running ldconfig")
logic-load                    | RuntimeError: can't find snap7 library. If installed, try running ldconfig

this is the version that gets installed
![image](https://github.com/gijzelaerr/python-snap7/assets/99563180/3cf6ad60-07a3-49cb-8d26-d3413aa75d7e)

**Reply by Rahul14189** [reply]
And my platform is 64 bit,

**Reply by nikteliy** [reply]
Is your issue resolved? Could you please share what the problem was?

**Reply by FernandoBarajasValdivia** [reply]
I could never solve it, to solve it I installed a virtual environment. So I was able to install the snap7 without problems. But witout the env I couldn't. :(

**Reply by FernandoBarajasValdivia** [reply]
It was so strange but I solved it.

**Reply by nikteliy** [reply]
Interesting. Thanks for sharing, this could be helpful. I will try to figure this out.

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. This was a native library loading issue. Python-snap7 3.0 is now a pure Python implementation and no longer requires the native snap7 library.
